diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index 8dd8845f3..c4b7e3754 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -3,7 +3,7 @@
 #
 # (based originally on the libLAS files copyright Mateusz Loskot)
 
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 
 #------------------------------------------------------------------------------
 # Configure build targets
diff --git a/cmake/dimension.cmake b/cmake/dimension.cmake
index ebb7222d4..69f2a7a20 100644
--- a/cmake/dimension.cmake
+++ b/cmake/dimension.cmake
@@ -4,7 +4,13 @@
 
 set(DIMENSION_INFILE ${PDAL_SRC_DIR}/Dimension.json)
 set(DIMENSION_OUTFILE ${CMAKE_CURRENT_BINARY_DIR}/include/pdal/Dimension.hpp)
+
+IF (CMAKE_CROSSCOMPILING)
+    SET(DIMBUILDER "${DIMBUILDER_EXECUTABLE}")
+ELSE (CMAKE_CROSSCOMPILING)
+    SET(DIMBUILDER dimbuilder)
+ENDIF (CMAKE_CROSSCOMPILING)
 add_custom_command(OUTPUT ${DIMENSION_OUTFILE}
-    COMMAND dimbuilder ${DIMENSION_INFILE} ${DIMENSION_OUTFILE}
+    COMMAND $DIMBUILDER ${DIMENSION_INFILE} ${DIMENSION_OUTFILE}
         DEPENDS ${DIMENSION_INFILE} dimbuilder)
 add_custom_target(generate_dimension_hpp DEPENDS ${DIMENSION_OUTFILE})
diff --git a/dimbuilder/CMakeLists.txt b/dimbuilder/CMakeLists.txt
index 09d334f1f..9aa9742f6 100644
--- a/dimbuilder/CMakeLists.txt
+++ b/dimbuilder/CMakeLists.txt
@@ -2,7 +2,7 @@
 # The seemingly strange arrangement of this cmake file allows you to build
 # this application as a standalone.
 #
-cmake_minimum_required(VERSION 2.8.6)
+cmake_minimum_required(VERSION 3.5)
 
 set(MAKE_COLOR_MAKEFILE ON)
 
diff --git a/scripts/conda/osx.sh b/scripts/conda/osx.sh
index 5c6badde0..b08342820 100755
--- a/scripts/conda/osx.sh
+++ b/scripts/conda/osx.sh
@@ -23,18 +23,13 @@ CFLAGS= CXXFLAGS= CC=/usr/bin/cc CXX=/usr/bin/c++ cmake   -G "$CONFIG"  \
         -DBUILD_PLUGIN_HDF=ON \
         -DBUILD_PLUGIN_ICEBRIDGE=ON \
         -DBUILD_PLUGIN_TILEDB=ON \
-        -DBUILD_PLUGIN_RDBLIB=ON \
-        -Drdb_DIR=/Users/hobu/dev/release/riegl/rdblib-2.2.3-x86_64-darwin/interface/cpp/ \
         -DWITH_LAZPERF=ON \
         -DWITH_ZSTD=ON \
-        -DWITH_LASZIP=ON \
-        -DWITH_DRACO=ON \
+        -DWITH_LASZIP=OFF \
+        -DBUILD_PLUGIN_DRACO=ON \
         ..
 
 
-# if ! [ -z "Unix Makefiles" ]; then
-#     make -j 4
-# fi
 
 if ! [ -z "Ninja" ]; then
     ninja
diff --git a/tools/lasdump/CMakeLists.txt b/tools/lasdump/CMakeLists.txt
index 8083a228d..767d642e7 100644
--- a/tools/lasdump/CMakeLists.txt
+++ b/tools/lasdump/CMakeLists.txt
@@ -2,7 +2,7 @@
 # The seemingly strange arrangement of this cmake file allows you to build
 # lasdump as a standalone.
 #
-cmake_minimum_required(VERSION 2.8.6)
+cmake_minimum_required(VERSION 3.5)
 
 set(MAKE_COLOR_MAKEFILE ON)
 
