diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6c940479..25bbcefde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -246,6 +246,7 @@ PDAL_ADD_LIBRARY(${PDAL_BASE_LIB_NAME} ${SRCS} ${RPLY_SRCS})
 # Interface include directories allow downstream project to get the directory
 # without specification.
 #
+
 target_include_directories(${PDAL_BASE_LIB_NAME}
     PRIVATE
         ${ROOT_DIR}
@@ -257,8 +258,8 @@ target_include_directories(${PDAL_BASE_LIB_NAME}
         ${ZSTD_INCLUDE_DIRS}
         ${NLOHMANN_INCLUDE_DIR}
     PUBLIC
-        ${GDAL_INCLUDE_DIR}
-        ${LASZIP_INCLUDE_DIR}
+        $<BUILD_INTERFACE:${GDAL_INCLUDE_DIR}>
+        $<BUILD_INTERFACE:${LASZIP_INCLUDE_DIR}>
 )
 target_link_libraries(${PDAL_BASE_LIB_NAME}
     PRIVATE
@@ -383,12 +384,14 @@ include(${PDAL_CMAKE_DIR}/config.cmake)
 # Allow downstream cmake projects to find PDAL header files without
 # being explicit.
 #
-target_include_directories(${PDAL_BASE_LIB_NAME}
-    INTERFACE
-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>)
-target_include_directories(${PDAL_UTIL_LIB_NAME}
-    INTERFACE
-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>)
+
+target_include_directories(${PDAL_LIB_NAME} INTERFACE
+    "$<BUILD_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>"
+    "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>")
+
+target_include_directories(${PDAL_UTIL_LIB_NAME} INTERFACE
+    "$<BUILD_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>"
+    "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>")
 
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES)
 
diff --git a/PDALConfig.cmake.in b/PDALConfig.cmake.in
index 237507f8e..f55974e25 100644
--- a/PDALConfig.cmake.in
+++ b/PDALConfig.cmake.in
@@ -4,16 +4,9 @@ set(PDAL_VERSION_MAJOR @PDAL_VERSION_MAJOR@)
 set(PDAL_VERSION_MINOR @PDAL_VERSION_MINOR@)
 set(PDAL_VERSION_PATCH @PDAL_VERSION_PATCH@)
 
-set(PDAL_INCLUDE_DIRS)
-set(PDAL_LIBRARY_DIRS)
-foreach(_dir @PDAL_CONFIG_INCLUDE_DIRS@)
-  set_and_check(_foo ${_dir})
-  list(APPEND PDAL_INCLUDE_DIRS ${_foo})
-endforeach(_dir)
-foreach(_dir @PDAL_CONFIG_LIBRARY_DIRS@)
-  set_and_check(_foo ${_dir})
-  list(APPEND PDAL_LIBRARY_DIRS ${_foo})
-endforeach(_dir)
+
+set_and_check(PDAL_INCLUDE_DIRS "@PACKAGE_INCLUDE_INSTALL_DIR@")
+set_and_check(PDAL_LIBRARY_DIRS "@PACKAGE_LIB_INSTALL_DIR@")
 
 include("${CMAKE_CURRENT_LIST_DIR}/PDALTargets.cmake")
 
diff --git a/cmake/config.cmake b/cmake/config.cmake
index f53a3498b..765a0027b 100644
--- a/cmake/config.cmake
+++ b/cmake/config.cmake
@@ -5,15 +5,15 @@ set(SYSCONFIG_INSTALL_DIR etc/pdal/ CACHE PATH "sysconfig")
 include(CMakePackageConfigHelpers)
 
 set(PDAL_CONFIG_INCLUDE_DIRS
-  "${CMAKE_INSTALL_PREFIX}/include")
+  "${PACKAGE_INSTALL_DIR}")
 set(PDAL_CONFIG_LIBRARY_DIRS
-  "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}")
+  "${PACKAGE_LIB_INSTALL_DIR}")
 
 configure_package_config_file(
   PDALConfig.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/PDALConfig.cmake
   INSTALL_DESTINATION ${LIB_INSTALL_DIR}/cmake/PDAL
-  PATH_VARS INCLUDE_INSTALL_DIR SYSCONFIG_INSTALL_DIR)
+  PATH_VARS INCLUDE_INSTALL_DIR SYSCONFIG_INSTALL_DIR LIB_INSTALL_DIR)
 
 write_basic_package_version_file(
   ${CMAKE_CURRENT_BINARY_DIR}/PDALConfigVersion.cmake
diff --git a/plugins/rdb/CMakeLists.txt b/plugins/rdb/CMakeLists.txt
index 7c0dc0bd4..86b80b0a0 100644
--- a/plugins/rdb/CMakeLists.txt
+++ b/plugins/rdb/CMakeLists.txt
@@ -15,7 +15,7 @@ PDAL_ADD_PLUGIN(libname reader rdb
         ${NLOHMANN_INCLUDE_DIR}
         ${RDB_INCLUDE_CPP_DIR}
 )
-target_include_directories(${libname} PUBLIC "${PDAL_VENDOR_DIR}/eigen")
+target_include_directories(${libname} PRIVATE "${PDAL_VENDOR_DIR}/eigen")
 
 # windows requires explicit linking against 
 # rdbcpp
