--- plugins/e57/CMakeLists.txt      2024-06-10 18:59:59.939789880 -0500
+++ plugins/e57/CMakeLists.txt      2024-06-10 19:00:34.831890623 -0500
@@ -18,6 +18,7 @@
         ${PDAL_VENDOR_DIR}
     LINK_WITH
         E57Format
+        ${PDAL_LIBRARIES}
     INCLUDES
         libE57Format/include
 )
@@ -34,6 +35,7 @@
         io/Uuid.cpp
     LINK_WITH
         E57Format
+        ${PDAL_LIBRARIES}
     INCLUDES
         libE57Format/include
 )
